src/components/TriggerDeliveryPolicyEditor.tsx

```diff
@@ -0,0 +1,677 @@
+/*
+This file is part of Astarte.
+
+Copyright 2023 SECO Mind Srl
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/* eslint-disable camelcase */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import { Button, Col, Container, Form, InputGroup, Modal, Row, Table } from 'react-bootstrap';
+import _, { toInteger } from 'lodash';
+import {
+  AstarteTriggerDeliveryPolicyDTO,
+  AstarteTriggerDeliveryPolicyHandlerDTO,
+} from 'astarte-client/types/dto/triggerDeliveryPolicy';
+import Icon from './Icon';
+
+type ErrorCodesControlProps = {
+  name: string;
+  value: string;
+  readonly: boolean;
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
```

Comment:
`ErrorCodesControl` is it's own component now, so if it's value is a `string`, when the value changes i would expect a `string`. Returning the whole input element leaks the inner working of the component to its user.

I would suggest using
```typescript
  onChange: (newValue: string) => void;
```
and adapt the input change handler
```typescript
  handleInputChange =
    (e: React.ChangeEvent<HTMLInputElement>) =>
      onChange(e.target.value)

  [...]

  onChange={handleInputChange}
```

This way from the outside you set a string, and you expect a string to be returned, hiding away how it's implemented